src/frontend/src/common/components/library/data_table/index.tsx

```diff
@@ -112,13 +151,45 @@ export const DataTable: FunctionComponent<Props> = ({
   handleHeaderCheckboxClick,
   handleRowCheckboxClick,
   isHeaderChecked,
+  setIsHeaderChecked,
   isHeaderIndeterminant,
+  setHeaderIndeterminant,
 }: Props) => {
   const [state, dispatch] = useReducer(reducer, {
     ascending: false,
     sortKey: defaultSortKey,
   });
 
+  useEffect(() => {
+    // used to determine if header is indeterminate
+    if (data) {
+      const publicIds = extractPublicIdsFromData(data, checkedSamples, true);
+      const sizeData = Object.keys(data).length;
+      if (publicIds.length === sizeData || publicIds.length === 0) {
+        setHeaderIndeterminant(false);
+      } else {
+        setHeaderIndeterminant(true);
+      }
+    }
+  }, [data, checkedSamples, setHeaderIndeterminant]);
+
+  useEffect(() => {
+    // used to determine if header is checked
```

Comment:
I feel mixed about suggesting this, but I'm wondering if you considered combining these two useEffect. They both begin with the same couple of lines of code that iterate through a potentially pretty large data object. They both also deal with checkbox state, and although checked vs indeterminant is not exactly the same thing, I think they are similar enough you _could_ combine them if you wanted.

No pressure. I could go either way on it. Wonder what you think.

Reply:
sure i combined them